new-system-tests/src/test/java/io/apicurio/registry/systemtests/SqlKeycloak.java

```diff
@@ -0,0 +1,108 @@
+        // PostgreSQL database resources
+        "/sql/00_service.yaml", // Service
+        "/sql/01_deployment.yaml", // Deployment
+        // Apicurio Registry operator
+        "/apicurio/00_operator_group.yaml", // Operator group for Apicurio Registry operator
+        "/apicurio/01_subscription.yaml", // Apicurio Registry operator subscription
+        // Apicurio Registry instance
+        "/apicurio/02_registry_sql_keycloak.yaml" // Apicurio Registry instance with PostgreSQL storage and Keycloak IAM
+})
+public class SqlKeycloak extends TestBase {
+    /** {@link OpenShiftClient} instance for tests. */
+    private OpenShiftClient client;
+
+    /**
+     * @return {@link OpenShiftClient} instance used in tests.
+     */
+    @Override
+    protected OpenShiftClient getClient() {
+        return client;
+    }
+
+    /**
+     * Performs actions that should be done before all tests.
+     */
+    @BeforeAll
+    public static void beforeAll() {
+        // Log information about current action
+        System.out.println("Before all.");
+
+        // Client is not accessible here
+    }
+
+    /**
+     * Performs actions that should be done after all tests.
+     */
+    @AfterAll
+    public static void afterAll() {
+        // Log information about current action
+        System.out.println("After all.");
+
+        // Client is not accessible here
+    }
+
+    /**
+     * Performs actions that should be done before each test.
+     */
+    @BeforeEach
+    public void beforeEach() {
+        // Log information about current action
+        System.out.println("Before each.");
+
+        // Wait for readiness of Keycloak deployment
+        assertTrue(waitDeploymentReady(client, Constants.KEYCLOAK_NAME));
+
+        // Wait for readiness of PostgreSQL database deployment
+        assertTrue(waitDeploymentReady(client, Constants.POSTGRESQL_NAME));
+
+        // Wait for readiness of Apicurio Registry operator deployment
+        assertTrue(waitDeploymentReady(client, findRegistryOperatorDeployment(client).getMetadata().getName()));
+    }
+
+    /**
+     * Performs actions that should be done after each test.
+     */
+    @AfterEach
+    public void afterEach() {
+        // Log information about current action
+        System.out.println("After each.");
+    }
+
+    /**
+     * Tests that {@link ApicurioRegistry} with PostgreSQL database storage and Keycloak IAM becomes ready.
+     */
+    @Test
+    public void testDeploy() {
+        // Log information about current action
+        System.out.println("### testDeploy test ###");
+
+        // Wait for readiness of Apicurio Registry instance with PostgreSQL database storage deployment
+        assertTrue(waitDeploymentReady(client, Constants.REGISTRY_NAME + "-deployment"));
+
+        // Check readiness of Apicurio Registry instance with PostgreSQL database storage deployment
+        assertTrue(isDeploymentReady(client, Constants.REGISTRY_NAME + "-deployment"));
```

Comment:
I'm not really sure what we are testing here, the `Deployment` might be `Ready` but since we are deploying also Keycloak we probably want to check that Registry and KC are working correctly together, isn't it?

Can you extend the test to verify an actual "user" scenario?

Reply:
This is very simple and basic test. It just tests that Apicurio Registry becomes ready when configured to use Keycloak for IAM service, nothing more.

I plan to add "real" authentication/authorization tests that will test Registry and KC integration later.

Reply:
will the "real" test substitute this one? if that's the case I'm good with it 👍

Reply:
Yes, I will substitute this "prototype" of test with "real" one in the following PR (tracked as https://github.com/Apicurio/apicurio-registry/issues/2993).